### ISSUE TITLE
Loading image shows incorrectly on Android

### ISSUE THREAD

**Issue by Wayward1** [minetest/minetest#5403]
The new loading bar image looks like this on my tablet:

![screenshot_2017-03-15-21-01-59](https://cloud.githubusercontent.com/assets/8821916/24012003/d3914ee8-0a52-11e7-8d05-a49e18815c08.png)

**Comment by kilbith** [comment]
I don't have a tablet for testing. Can you try that patch and see if it fixes your issue ?

```Diff
diff --git a/src/drawscene.cpp b/src/drawscene.cpp
index e3e6301a..8bf03092 100644
--- a/src/drawscene.cpp
+++ b/src/drawscene.cpp
@@ -615,8 +615,7 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
 			u32 imgH = rangelim(img_size.Height, 24, 72);
 			v2s32 img_pos((screensize.X - imgW) / 2, (screensize.Y - imgH) / 2);
 
-			draw2DImageFilterScaled(
-				driver, progress_img_bg,
+			driver->draw2DImage(progress_img_bg,
 				core::rect<s32>(img_pos.X,
 						img_pos.Y,
 						img_pos.X + imgW,
@@ -626,8 +625,7 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
 						img_size.Height),
 				0, 0, true);
 
-			draw2DImageFilterScaled(
-				driver, progress_img,
+			driver->draw2DImage(progress_img,
 				core::rect<s32>(img_pos.X,
 						img_pos.Y,
 						img_pos.X + (percent * imgW) / 100,
```

**Comment by kilbith** [comment]
@SmallJoker is also suggesting this patch : 

```Diff
diff --git a/src/drawscene.cpp b/src/drawscene.cpp
index e3e6301a..8fdb1ac4 100644
--- a/src/drawscene.cpp
+++ b/src/drawscene.cpp
@@ -610,7 +610,7 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
 			driver->getTexture(tp_progress_bar_bg.c_str());
 
 		if (progress_img && progress_img_bg) {
-			const core::dimension2d<u32> &img_size = progress_img_bg->getSize();
+			const core::dimension2d<u32> &img_size = progress_img_bg->getOriginalSize();
 			u32 imgW = rangelim(img_size.Width, 200, 600);
 			u32 imgH = rangelim(img_size.Height, 24, 72);
 			v2s32 img_pos((screensize.X - imgW) / 2, (screensize.Y - imgH) / 2);
```

**Comment by Wayward1** [comment]
Thank you for the quick response; unfortunately, neither worked.

**Comment by kilbith** [comment]
All images on OpenGL ES < 2.0 are scaled on a size of a power of 2 ([NPOT textures extension](https://www.khronos.org/opengl/wiki/NPOT_Texture)). The current progress bar is 256x48 px, try resizing the images at 256x64 px and see if it looks good.

For devs : http://stackoverflow.com/questions/3740077/can-opengl-es-render-textures-of-non-base-2-dimensions

**Comment by kilbith** [comment]
Can you try that patch please ? (against current git HEAD).

```Diff
diff --git a/src/drawscene.cpp b/src/drawscene.cpp
index e3e6301a..ee8ffdf7 100644
--- a/src/drawscene.cpp
+++ b/src/drawscene.cpp
@@ -17,6 +17,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#include <GL/glew.h>
 #include "drawscene.h"
 #include "settings.h"
 #include "clouds.h"
@@ -32,6 +33,20 @@ typedef enum {
 	EYECOUNT = 2
 } paralax_sign;
 
+inline u16 get_GL_major_version()
+{
+	const GLubyte *gl_version = glGetString(GL_VERSION);
+	std::string gl_ver((const char *)gl_version);
+	u16 GL_version = stoi(gl_ver.substr(0,1));
+	return GL_version;
+}
+
+inline u32 get_POT_size(const u32 &size)
+{
+	u32 new_size = pow(2, ceil(log(size) / log(2)));
+	return new_size;
+}
+
 void draw_anaglyph_3d_mode(Camera& camera, bool show_hud, Hud& hud,
 		video::IVideoDriver* driver, scene::ISceneManager* smgr,
 		bool draw_wield_tool, Client& client,
@@ -613,6 +628,10 @@ void draw_load_screen(const std::wstring &text, IrrlichtDevice* device,
 			const core::dimension2d<u32> &img_size = progress_img_bg->getSize();
 			u32 imgW = rangelim(img_size.Width, 200, 600);
 			u32 imgH = rangelim(img_size.Height, 24, 72);
+
+			imgW = get_GL_major_version() > 1 ? imgW : get_POT_size(imgW);
+			imgH = get_GL_major_version() > 1 ? imgH : get_POT_size(imgH);
+
 			v2s32 img_pos((screensize.X - imgW) / 2, (screensize.Y - imgH) / 2);
```

**Comment by sfan5** [comment]
@kilbith 
just put it into an `#ifdef __ANDROID__` and maybe reuse the function from [tile.cpp](https://github.com/minetest/minetest/blob/master/src/client/tile.cpp#L1063)

**Comment by kilbith** [comment]
@sfan5 
The issue is with OpenGL/ES 1.x only.

**Comment by Wayward1** [comment]
Sorry, I've been busy with school and other IRL stuff. I'll see if I can try it tomorrow

**Comment by sfan5** [comment]
@kilbith Indeed, however all the other code also assumes that android always needs the fix.

**Comment by nerzhul** [comment]
tested last patch, no effect

**Comment by nerzhul** [comment]
Fixed by #5601
